uint8 output1 # control IO
uint8 output2
uint8 output3
uint8 output4
uint8 output5
uint8 output6